pkg/vault/envconsul/webhook.go

```diff
@@ -0,0 +1,196 @@
+
+	for k, container := range mutatedPod.Spec.Containers {
+		configPath, ok := containersConfigPaths[container.Name]
+		if !ok {
+			podLogger.Log("event", "container.skipped", "msg", fmt.Sprintf("container not configured: %s", container.Name))
+			continue
+		}
+
+		c := &mutatedPod.Spec.Containers[k]
+
+		originalCommand := strings.Join(append(c.Command, c.Args...), " ")
+		c.Command = buildCommand(originalCommand, installPath, configPath, vaultAddress, vaultAuthMountPath, vaultAuthRole)
+
+		c.VolumeMounts = append(c.VolumeMounts, buildVolumeMount(installPath, true))
+
+		podLogger.Log("event", "container.configured", "msg", fmt.Sprintf("container configured: %s", container.Name))
+	}
+
+	return admission.PatchResponse(pod, mutatedPod)
+}
+
+func getContainersConfigPaths(config string) map[string]string {
+	containersConfigPaths := map[string]string{}
+
+	containers := strings.Split(config, ",")
+	for _, container := range containers {
+		nameAndPath := strings.SplitN(container, ":", 2)
+		name := strings.TrimSpace(nameAndPath[0])
+		path := ""
+		if len(nameAndPath) == 2 {
+			path = strings.TrimSpace(nameAndPath[1])
+		}
+		containersConfigPaths[name] = path
+	}
+
+	return containersConfigPaths
+}
+
+func buildCommand(originalCommand, installPath, configPath, vaultAddress, vaultAuthMountPath, vaultAuthRole string) []string {
+	command := []string{
+		path.Join(installPath, "theatre-envconsul"),
+		"exec",
+		fmt.Sprintf("--vault-address=%s", vaultAddress),
+	}
+
+	if installPath != "" {
+		command = append(command, fmt.Sprintf("--install-path=%s", installPath))
+	}
+	if vaultAuthMountPath != "" {
+		command = append(command, fmt.Sprintf("--auth-backend-mount-path=%s", vaultAuthMountPath))
+	}
+	if vaultAuthRole != "" {
+		command = append(command, fmt.Sprintf("--auth-backend-role=%s", vaultAuthRole))
+	}
+	if configPath != "" {
+		command = append(command, fmt.Sprintf("--config-file=%s", configPath))
+	}
+	command = append(command, fmt.Sprintf("--command=%s", originalCommand))
+
+	return command
+}
+
+func buildInitContainer(image, path string) corev1.Container {
+	return corev1.Container{
+		Name:    "init-vault",
+		Image:   image,
+		Command: []string{"/theatre-envconsul", "install", "--path", path},
+		VolumeMounts: []corev1.VolumeMount{
+			buildVolumeMount(path, false),
+		},
+		Resources: corev1.ResourceRequirements{
+			Limits: corev1.ResourceList{
+				corev1.ResourceMemory: resource.MustParse("64Mi"),
+				corev1.ResourceCPU:    resource.MustParse("50m"),
+			},
+			Requests: corev1.ResourceList{
+				corev1.ResourceMemory: resource.MustParse("64Mi"),
+				corev1.ResourceCPU:    resource.MustParse("50m"),
+			},
+		},
+	}
+}
+
+func buildVolumeMount(path string, readOnly bool) corev1.VolumeMount {
+	return corev1.VolumeMount{
+		Name:      "vault",
+		MountPath: path,
+		ReadOnly:  readOnly,
+	}
+}
+
+func buildVolume() corev1.Volume {
+	return corev1.Volume{
+		Name: "vault",
```

Comment:
We should be super careful with these names and make sure they are very explicit. Otherwise we run the risk of colliding with people legitimately creating volume mounts.

This may well collide with our vault statefulsets, for example. And that collision would be confusing and disruptive.